### PR TITLE
Fix challenge title formatting misc

### PR DESCRIPTION
**Pull request by dhcodes**
#### Pre-Submission Checklist

- [X] Your pull request targets the `staging` branch of freeCodeCamp.
- [X] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [X] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [X] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [X] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [X] Tested changes locally.
- [ ] Closes currently open issue (replace XXXX with an issue no): Part of #12698

#### Description

Made a lot of formatting changes to challenge titles, including:
- Capitalize 'Your'
- lowercased attributes, tags, and method names in titles
- (Where possible) Remove 'ing' from end of verb in challenge title--did not do this on Basic JS section since we'd also have to change the Wiki to reflect these.
- Add to title to discuss what a method or function does
- Misc. capitalization or punctuation.

**TLDR;** I tried to make everything as uniform as possible.

### CODE REVIEW

**Review by Greenheart — CHANGES_REQUESTED** [review]
@dhcodes Massive update! Thanks :smile: :+1:

File: seed/challenges/02-javascript-algorithms-and-data-structures/es6.json [hunk 1]
```diff
@@ -281,7 +281,7 @@
     },
     {
       "id": "587d7b88367417b2b2512b46",
-      "title": "Write Functions with Default Parameters",
+      "title": "Set Default Parameters on Your Functions",
```
Comment (dhcodes) [inline seed/challenges/02-javascript-algorithms-and-data-structures/es6.json]: Good catch. I'll fix this and rebase it Monday.

**Comment by dhcodes** [comment]
Okay, I think this is good to go now. Yay for resolving merge conflicts while wondering the whole time "am i doing this right?".

**Review by Greenheart — APPROVED** [review]
@dhcodes Thanks! I know that feeling all too well! :smile:
